File: js/videoStreamingPlans.js
// Given a class for a BasicPlan, write the classes for StandardPlan and PremiumPlan which have class properties of the following:
// BasicPlan	StandardPlan	PremiumPlan
//    ✓	            ✓	          ✓	      canStream
//    ✓	            ✓	          ✓	   		canDownload
//    ✓	            ✓	          ✓	   		hasSD
//                   ✓	         ✓	 		 hasHD
//    													✓		 	 	 hasUHD
// 		1							 2					 4			 numOfDevices
//  $8.99					$12.99				$15.99	 price

class BasicPlan {
  static canStream = true;
  static canDownload = true;
  static hasSD = true;
  static hasHD = false;
  static hasUHD = false;
  static numOfDevices = 1;
  static price = "$8.99";
}

class StandardPlan extends BasicPlan {
  static hasHD = true;
  static hasUHD = false;
  static numOfDevices = 2;
  static price = "$12.99";
}

class PremiumPlan extends StandardPlan {
  static hasUHD = true;
  static numOfDevices = 4;
  static price = "$15.99";
}

module.exports = {
  BasicPlan,
  StandardPlan,
  PremiumPlan
};

File: python/video_streaming_plans.py
"""
Video Streaming Plans

Given a class for a BasicPlan, write the classes for StandardPlan and PremiumPlan which have class attributes of the following:

BasicPlan	StandardPlan	Premium Plan
    ✓	         ✓	            ✓	    can_stream
    ✓	         ✓	            ✓	    can_download
    ✓	         ✓	            ✓	    has_SD
                 ✓	            ✓	     has_HD
                                ✓	     has_UHD
    1	         2	             4	      num_of_devices
    $8.99	   $12.99	       $15.99	  price

https://edabit.com/challenge/5T978H873HFZ7xKd8
"""


class BasicPlan:
    can_stream = True
    can_download = True
    has_SD = True
    has_HD = False
    has_UHD = False
    num_of_devices = 1
    price = '$8.99'


class StandardPlan(BasicPlan):
    has_HD = True
    num_of_devices = 2
    price = '$12.99'


class PremiumPlan(StandardPlan):
    has_UHD = True
    num_of_devices = 4
    price = '$15.99'
